Rb.Forms.Barcode.iOS/BarcodeScannerRenderer.cs

```diff
@@ -0,0 +1,206 @@
+            session.StartRunning();
+            captureVideoPreviewLayer.Orientation = (AVCaptureVideoOrientation) UIApplication.SharedApplication.StatusBarOrientation;
+            this.Debug("StartRunning");
+            addOrientationObserver();
+        }
+
+        private void stopSession() 
+        {
+            session.StopRunning();
+            this.Debug("StopRunning");
+            removeOrientationObserver();
+        }
+
+        private void addOrientationObserver()
+        {
+            orientationObserverToken = NSNotificationCenter.DefaultCenter.AddObserver(UIDevice.OrientationDidChangeNotification, updateViewOnOrientationChanged);
+        }
+
+        private void removeOrientationObserver()
+        {
+            NSNotificationCenter.DefaultCenter.RemoveObserver(orientationObserverToken);
+        }
+
+        private void updateSize(PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == VisualElement.WidthProperty.PropertyName || e.PropertyName == VisualElement.HeightProperty.PropertyName) {
+                captureVideoPreviewLayer.Frame = new CGRect(0, 0, Element.Width, Element.Height);
+                this.Debug("Preview Frame changed to {0}", captureVideoPreviewLayer.Frame);
+            }
+        }
+
+        private void updateTorch(PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == BarcodeScanner.TorchProperty.PropertyName && device.TorchAvailable) {
+                NSError error;
+                device.LockForConfiguration(out error);
+
+                if (error != null) {
+                    this.Debug("Torch error");
+
+                    return;
+                }
+
+                device.TorchMode = barcodeScanner.Torch ? AVCaptureTorchMode.On : AVCaptureTorchMode.Off;
+
+                device.UnlockForConfiguration();
+                this.Debug("Torch AVCaptureTorchMode {0}", barcodeScanner.Torch);
+            }
+        }
+
+        private bool initScanner()
+        {
+            device = AVCaptureDevice.DefaultDeviceWithMediaType(AVMediaType.Video);
+            if (device == null) {
+                this.Debug("AVCaptureDeviceInput is null");
+
+                return false;
+            }
+
+            input = AVCaptureDeviceInput.FromDevice(this.device);
+            session = new AVCaptureSession();
+            session.AddInput(input);
+
+            output = new AVCaptureMetadataOutput();
+            output.SetDelegate(this, DispatchQueue.MainQueue);
+
+            session.AddOutput(output);
+            output.MetadataObjectTypes = configuraiton.Barcodes.ConvertToIOS();
+
+            captureVideoPreviewLayer = AVCaptureVideoPreviewLayer.FromSession(session);
+            captureVideoPreviewLayer.Frame = CGRect.Empty;
+            captureVideoPreviewLayer.VideoGravity = AVLayerVideoGravity.ResizeAspectFill;
+            captureVideoPreviewLayer.Orientation = (AVCaptureVideoOrientation) UIApplication.SharedApplication.StatusBarOrientation;
```

Comment:
I know about this deprecation. but i thourgt that this was marked as deprecated in version ios 8. we want to use ios7 so i let that deprecation in. but when it is already deprecated in ios6 it makes sense to change it.